src/status_im2/contexts/profile/login/events.cljs

```diff
@@ -181,3 +182,32 @@
   (rf/merge cofx
             (navigation/init-root :profiles)
             (biometric/show-message code)))
+
+(rf/defn verify-database-password
+  {:events [:profile.login/verify-database-password]}
+  [{:keys [db] :as cofx} entered-password]
+  (let [hashed-password (-> entered-password
+                            security/safe-unmask-data
+                            ethereum/sha3)
+        key-uid         (get-in db [:profile/login :key-uid])
+        valid?          (native-module/verify-database-password
+                         key-uid
+                         hashed-password
+                         (fn [response]
+                           (let [valid? (string/blank? (.-error (js/JSON.parse response)))]
+                             (rf/dispatch [:profile.login/verified-database-password valid?]))))]))
+
+(rf/defn verify-database-password-success
+  {:events [:profile.login/verified-database-password]}
+  [{:keys [db] :as cofx} valid?]
+  (if valid?
+    (do
+      {:db (update db
+                   :profile/login
+                   dissoc
+                   :processing :error)})
+    {:db (update db
+                 :profile/login
+                 #(-> %
+                      (dissoc :processing)
+                      (assoc :error "Invalid password")))}))
```

Comment:
I believe we should use `(i18n/label :t/oops-wrong-password)`

Reply:
sure, I can make that adjustment. This is just the error message passed to the db here. There is some weird hack on the display of actually showing that. Actually I wonder are we okay with adding i18n messages to the db? @cammellos is this okay for us or should we keep it english only in the db? 🤔

Reply:
Just raised that comment from a UX standpoint. Because we can't display English if a user uses the app in a different language. 👍 

(There are quite a few usages of `i18n` in reframe event handlers in our codebase. Not sure if it's allowed)

Reply:
yep, this is currently how the password error message is working 😶
https://github.com/status-im/status-mobile/blob/50625c5bde2c208a22e3b5aad85ec486ff851eea/src/status_im2/common/standard_authentication/password_input/view.cljs#L13-L20
I just moved that code but I believe this should have a stronger mechanism - and I think better as you are saying the error message ideally is just set appropriately where the error originates.

Reply:
Maybe the best is to treat the _translated text_ as part of the view layer, and the _translation keys_ the data that can be persisted in the app db. This way, the db layer stays translation agnostic. Not a big deal though, but seems better, also facilitates unit testing.